src/components/DayPicker.jsx

```diff
@@ -395,8 +404,12 @@ export default class DayPicker extends React.Component {
       this.props.onNextMonthClick(e);
     }
 
-    const translationValue =
-      this.isVertical() ? -this.getMonthHeightByIndex(1) : -this.dayPickerWidth;
+    const convertedTranslationValueRTL = (this.dayPickerWidth + this.state.translationValueRTL);
+    let translationValue = this.props.isRTL ? convertedTranslationValueRTL : -this.dayPickerWidth;
+
+    if (this.isVertical()) {
+      translationValue = this.getMonthHeightByIndex(1) * -1;
```

Comment:
the hard-coded 1 wasn't written by me., it's from the master:
![rd](https://cloud.githubusercontent.com/assets/9153149/25303109/da7a0b4a-2754-11e7-8b2e-e0d3f70e172b.png)